dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt

```diff
@@ -143,23 +159,98 @@ class DataManagerTest(
 
     @Test
     fun `check that an exception is thrown if the received data from the storage has an unexpected uploading user`() {
-        val storableDataSet = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinacialsForIt()
-        val storableDataSetAsString = objectMapper.writeValueAsString(storableDataSet)
-        `when`(mockStorageClient.insertData(correlationId, storableDataSetAsString)).thenReturn(
+        val storableDataSetForNonFinancials = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinancialsForIt()
+        val messageForDataSetAndCorrelationId = buildDummyMessage(storableDataSetForNonFinancials)
+        `when`(mockStorageClient.insertData(messageForDataSetAndCorrelationId)).thenReturn(
             InsertDataResponse(dataUUId),
         )
-        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        val dataId = dataManager.addDataSet(storableDataSetForNonFinancials, correlationId)
 
         `when`(mockStorageClient.selectDataById(dataId, correlationId)).thenReturn(
-            objectMapper.writeValueAsString(storableDataSet.copy(uploaderUserId = "NOT_WHATS_EXPECTED")),
+            buildReturnOfMockDataSelect(storableDataSetForNonFinancials),
         )
 
         val thrown = assertThrows<InternalServerErrorApiException> {
-            dataManager.getDataSet(dataId, storableDataSet.dataType, correlationId)
+            dataManager.getDataSet(dataId, storableDataSetForNonFinancials.dataType, correlationId)
         }
         assertEquals(
-            "The meta-data of dataset $dataId differs between the data store and the database",
-            thrown.message,
+            "The meta-data of dataset $dataId differs between the data store and the database", thrown.message,
+        )
+    }
+    private fun buildReturnOfMockDataSelect(storableDataSetForNonFinancials: StorableDataSet): String {
+        return objectMapper.writeValueAsString(
+            storableDataSetForNonFinancials.copy(
+                uploaderUserId = "NOT_WHATS_EXPECTED",
+            ),
```

Comment:
why copying the original object - why not just updating it?

Reply:
I think this is due to the uploaderUserId of the StorableDataSet being type val and therefor it could not be updated directly.

Reply:
well - make it var?

Reply:
Would be a possibility, however do we want the uploader of a dataset to be changeble? I thought it makes somehow sense that this should not be changed once it was set for the first time, or do i miss something here?

Reply:
I'm agree with Stephan.

Reply:
OK